addons/stock/models/stock_inventory.py

```diff
@@ -164,6 +170,46 @@ def action_reset_product_qty(self):
         return True
 
     def action_validate(self):
+        # Check the serial numbers uniquiness
+        serials = self.line_ids.filtered(lambda l: l.product_id.tracking == 'serial' and l.prod_lot_id and l.theoretical_qty < l.product_qty)
+        wiz_lines = []
+        for line in serials:
+            similar_quant = self.env['stock.quant'].search([
+                ('lot_id', '=', line.prod_lot_id.id),
+                ('location_id', '!=', line.location_id.id),
+                '|',
+                ('location_id.usage', 'not in', ('supplier', 'inventory', 'production')),
+                ('location_id.scrap_location', '=', True),
+                ('quantity', '>', 0)
```

Comment:
I would rather remove the quantity from the search and do a sim instead of using [0]